src/Agent.Plugins/Artifact/PipelineArtifactServer.cs

```diff
@@ -69,7 +78,16 @@ public PipelineArtifactServer(IAppTraceSource tracer)
 
                 // 2) associate the pipeline artifact with an build artifact
                 BuildServer buildServer = new BuildServer(connection);
-                Dictionary<string, string> propertiesDictionary = new Dictionary<string, string>();
+
+                // Prevent custom properties from overwriting computed ones below
```

Comment:
I'd prefer if instead of trying to manually integrate our properties with user-supplied properties we added a new "UserProperties" key (or something similar, naming is hard) and just shoved the user-supplied JSON there.  That key would then be reserved for that use case and we wouldn't have to worry about collisions like this.

Reply:
As discussed, updating logic to ensure all custom properties have "user-" prefix.